src/gen9_vdenc.c

```diff
@@ -3670,13 +3670,13 @@ gen9_vdenc_avc_encode_picture(VADriverContextP ctx,
         gen9_vdenc_mfx_vdenc_pipeline(ctx, encode_state, encoder_context);
         gen9_vdenc_read_status(ctx, encoder_context);
 
-        intel_batchbuffer_end_atomic(batch);
-        intel_batchbuffer_flush(batch);
-
         vdenc_context->brc_initted = 1;
         vdenc_context->brc_need_reset = 0;
     }
 
+    intel_batchbuffer_end_atomic(batch);
+    intel_batchbuffer_flush(batch);
+
     return VA_STATUS_SUCCESS;
 }
 
```

Comment:
Multi-pass scenario is also considered and there is no problem.
This patch is only for the optimization purpose.  Previously, maybe multi drm_ioctls are needed.
After it is combinted, only one drm_ioctl is needed.